File: internal/infra/aws/sm_client.go
package infra_aws

import (
	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
)

type SMClient struct {
	*secretsmanager.Client
}

func NewSMClient(region AwsRegion, stage AwsStage) *SMClient {
	if stage != StageLocal {
		return &SMClient{
			Client: secretsmanager.NewFromConfig(NewConfig(region)),
		}
	}

	return &SMClient{
		Client: secretsmanager.NewFromConfig(NewLocalConfig(region)),
	}
}

File: src/infra/aws/sm_client/sm_client.ts
import { SecretsManagerClient } from "@aws-sdk/client-secrets-manager"
import { AwsRegion, AwsStage, newAwsConfig } from "../utils"

export class SMClientWrapper extends SecretsManagerClient {
	constructor(region: AwsRegion, stage: AwsStage) {
		const config = newAwsConfig(region, stage)
		super(config)
	}
}